syntax = "proto3";

import "google/protobuf/any.proto";
import "attackType.proto";

message GameCommand {
    google.protobuf.Any command = 1;
}

message CommandGetGameInfo {}

message CommandSetDeck {
    string deck = 1;
}

message CommandReadyToStart {
    bool ready = 1;
}

message CommandDeclareAttack {
    sint32 stage_pos = 1;
    AttackType attack_type = 2;
}

message CommandLevelUp {
    sint32 clock_pos = 1;
}

message CommandEncoreCharacter {
    sint32 stage_pos = 1;
}

message CommandEndTurn {}
